reminders.go

```diff
@@ -0,0 +1,45 @@
+package slack
+
+import (
+	"context"
+	"errors"
+	"net/url"
+)
+
+// AddReminder adds a reminder for a channel or a user.
+//
+// Only one of ChannelID or UserID should be provided. Both can be omitted, in
+// which case the reminder is set for the authenticated user.
+//
+// See https://api.slack.com/methods/reminders.add (NOTE: the ability to set
+// reminders on a channel is currently undocumented but has been tested to
+// work)
+func (api *Client) AddReminder(channelID, userID, text, time string) error {
+	return api.AddReminderContext(context.Background(), channelID, userID, text, time)
+}
+
+// AddReminderContext adds a reminder for a channel or a user with a custom context.
+//
+// See AddReminder for full details.
+func (api *Client) AddReminderContext(ctx context.Context, channelID, userID, text, time string) error {
```

Comment:
can you have a channel and a user simultaneously? otherwise thoughts on just having a methods for each case?